### ISSUE TITLE
libtorch_python.so: undefined symbol: PyInstanceMethod_Type

### ISSUE THREAD

**Issue by jjsjann123** [pytorch/pytorch#90016]
### 🐛 Describe the bug

I run into this error vvv

```
C++ exception with description "/var/lib/jenkins/multipy/multipy/runtime/loader.cpp:780: libnvfuser_python.so: could not load library, dlopen says: /opt/conda/lib/python3.10/site-packages/torch/lib/libtorch_python.so: undefined symbol: PyInstanceMethod_Type" thrown in the test body.
```

link of failing CI: https://github.com/pytorch/pytorch/actions/runs/3586511045/jobs/6036274959
cmake lines related to the object: https://github.com/pytorch/pytorch/blob/d8335704a54f89e93f562cda7b49048e1eab1ebb/torch/csrc/jit/codegen/cuda/CMakeLists.txt#L165-L200
It's a little strange to me how the error was thrown from libtorch_python.so. This feels like a pytorch build issue.

I'm also a little bit uncertain how this is invoked.

In the setup, I have `libnvfuser_python.so` which depends on both `libnvfuser_codegen.so` and `libtorch_python.so`. While `libnvfuser_codegen.so` is dlopened by torch and does not have dependency on `libnvfuser_python.so`.

### Versions

This is failing on CI job:

[linux-bionic-cuda11.6-py3.10-gcc7 / test (deploy, 1, 1, linux.4xlarge.nvidia.gpu)](https://github.com/pytorch/pytorch/actions/runs/3590997095/jobs/6045485004)

cc @malfet @seemethere

**Comment by jjsjann123** [comment]
One thing I did notice is that for linux-bionic-cuda11.6-py3.10-gcc7 

CI build uses `Found PythonLibs: /opt/conda/lib/libpython3.10.a (found version "3.10.4")`
and tests runs uses 3.10.8
```
2022-11-30T20:20:41.8143520Z     libpython-static-3.10.8    |       h6a678d5_0        54.4 MB
2022-11-30T20:20:41.8144242Z     python-3.10.8              |       haa1d7c7_0        26.8 MB
```

But minor version should be compatible...

**Comment by jjsjann123** [comment]
Bumping threads~
since the `libtorch_python.so` issue is the only failing test on my PR. :cry:

**Comment by jjsjann123** [comment]
bumping threads again:

Just realized that the failing config is `deploy` and the test somehow is dlopen `libnvfuser_python.so`. which has python dependency and likely that's what's missing there?!?!
I'm not sure where those tests are defined, and why are they explicitly loading `libnvfuser_python.so`?
```
[  FAILED  ] 20 tests, listed below:
[  FAILED  ] TorchpyTest.InitManagerBasic
[  FAILED  ] TorchpyTest.InitTwice
[  FAILED  ] TorchpyTest.DifferentInterps
[  FAILED  ] TorchpyTest.SimpleModel
[  FAILED  ] TorchpyTest.ResNet
[  FAILED  ] TorchpyTest.Movable
[  FAILED  ] TorchpyTest.MultiSerialSimpleModel
[  FAILED  ] TorchpyTest.ThreadedSimpleModel
[  FAILED  ] TorchpyTest.ErrorsReplicatingObj
[  FAILED  ] TorchpyTest.ThrowsSafely
[  FAILED  ] TorchpyTest.AcquireMultipleSessionsInTheSamePackage
[  FAILED  ] TorchpyTest.AcquireMultipleSessionsInDifferentPackages
[  FAILED  ] TorchpyTest.TensorSharingNotAllowed
[  FAILED  ] TorchpyTest.TaggingRace
[  FAILED  ] TorchpyTest.DisarmHook
[  FAILED  ] TorchpyTest.RegisterModule
[  FAILED  ] TorchpyTest.TensorSerializationSharing
[  FAILED  ] TorchpyTest.UsesDistributed
[  FAILED  ] TorchpyTest.Autograd
[  FAILED  ] TorchpyTest.PrintInstruction
```

**Comment by crcrpar** [comment]
from https://github.com/pytorch/pytorch/blob/797544f1c456035db1d3fbbd9141d592a834c60a/.jenkins/pytorch/common_utils.sh#L147-L169 the multipy loader.cpp seems to be https://github.com/pytorch/multipy/blob/60d9ef2df2950eb7cbaed671eb1fe4eb1e02525b/multipy/runtime/loader.cpp which meticulously handles `libtorch_python.so` https://github.com/pytorch/multipy/blob/60d9ef2df2950eb7cbaed671eb1fe4eb1e02525b/multipy/runtime/loader.cpp#L720-L723

**Comment by jjsjann123** [comment]
Thx for that. I would need to add an entry to skip libnvfuser_python.so as well on deploy.

**Comment by jjsjann123** [comment]
I think the issue here is that we are loading libnvfuser_python.so because it's a python extension under `torch._C_nvfuser`. I think there's something I did wrong, that I'm not handling the load properly. I don't understand how this is supposed to work in the deploy tests... I guess I should magically strip libtorch_python.so dependency for the test to pass, but that doesn't work for nvfuser python interface to work independently.

Anyway, I decided to work towards a different direction and take nvfuser out of torch package. This worked in my PR :tada: 

I'm closing this one as it is a user error.
